recipes/dakota/meta.yaml

```diff
@@ -0,0 +1,77 @@
+{% set name = "dakota" %}
+{% set version = "6.10" %}
+{% set sha256sum = "d989f77577199497dfcd49e4b13b85cb7dbe20e82fc28a15b4d3811d9e362f21" %}
+
+package:
+    name: {{ name }}
+    version: {{ version }}
+
+source:
+    fn: dakota{{ version }}.tar.gz
+    url: https://dakota.sandia.gov/sites/default/files/distributions/public/dakota-{{ version }}-release-public.src.tar.gz
+    sha256: {{ sha256sum }}
+    patches:
+        - surfpack_py_bindings.patch
+        - gsl.patch
+        - python.patch
+        - colin_boost170.patch
+
+build:
+    number: 0
+    skip: true  # [win or osx or py<37]
+
+requirements:
+    build:
+        - cmake
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - {{ compiler('fortran') }}
+    host:
+        - openmpi
+        - pybind11
+        - boost-cpp
+        - libblas
+        - libcblas
+        - python
+        - perl
+        - numpy
+        - swig
+        - doxygen
+        - gsl
+        # - trilinos  # needs newer trilinos
+    run:
+        - openmpi
+        - boost-cpp
+        - python
+        - {{ pin_compatible("numpy") }}
+        - nomkl
+        - openblas
+        - gsl
+        # - trilinos
+test:
+    imports:
+        - dakota
+        - dakota.interfacing
+        - dakota.interfacing.dprepro
+        - pysurfpack
```

Comment:
Does this vend pysurfpack? Or is that wholly developed as part of dakota?

Reply:
Dakota provides mainly an interface to many different scientific libraries (Trilinos, pecos, surfpack, ...) These libraries are included as git subrepos and the source is directly included in the release. Ideally every single library would have it's own conda-package. But this will increase the maintenance-difficulties as already an external trilionos is in conflict with this recipe.

Pysurfpack is a python binding to surfpack which is part of dakota. Pysurfpack are only a few wrapper files which are added by a patch and are not part of dakota.

Reply:
This seems really weird.  Can you provide a link to what is and is not part of dakota.  If something is developed separately, it needs its own conda-forge package.

Reply:
> If something is developed separately, it needs its own conda-forge package

I think trilinos is similar and it's also packages for conda-forge. 

> This seems really weird. Can you provide a link to what is and is not part of dakota.

Best to look at the cmake file. <coming soon>

Reply:
So all of those should be other conda-forge packages then?

Reply:
I am not really sure. I guess keeping them compatible is easier by using source from dakota directly. Also I am not sure if any other project is using one of these libraries.

Reply:
Well, if they are independently developed (ie in separate repos or with separate tarballs) then they should be sepearte packages

Reply:
I guess this will be difficult, as the dakota release is more a bundle of compatible sources. Packaging strategy of dakota  seems to be a bit different (compared to conda-forge). While I would love to see all packages being added separately, I don't think this is feasible.

Reply:
Ok, well, surfpack is only developed in dakota (https://dakota.sandia.gov/content/surfpack) which was my question. So this is fine.